include/boost/geometry/strategies/spherical/area_huiller.hpp

```diff
@@ -142,20 +150,14 @@ public :
 
             E = geometry::math::abs(E);
 
-            // In right direction: positive, add area. In left direction: negative, subtract area.
-            // Longitude comparisons are not so obvious. If one is negative, other is positive,
-            // we have to take the dateline into account.
-            // TODO: check this / enhance this, should be more robust. See also the "grow" for ll
-            // TODO: use minmax or "smaller"/"compare" strategy for this
-            calculation_type lon1 = geometry::get_as_radian<0>(p1) < 0
-                ? geometry::get_as_radian<0>(p1) + two_pi
-                : geometry::get_as_radian<0>(p1);
-
-            calculation_type lon2 = geometry::get_as_radian<0>(p2) < 0
-                ? geometry::get_as_radian<0>(p2) + two_pi
-                : geometry::get_as_radian<0>(p2);
+            calculation_type lon = geometry::get_as_radian<0>(p2)
+                                 - geometry::get_as_radian<0>(p1);
+            if ( lon <= 0 )
```

Comment:
Although it does not make much of a difference, I would write this as `lon < 0`. If `lon` is 0 then the segment line along a meridian and the area of the triangle formed by `p1`, `p2` and the south pole is identically equal to zero.

My suggestions would be:

1. Check if the two longitudes are the same. Then in this case do not compute the excess, but rather set it directly to 0.
2. Use `math::smaller(lon, 0)` rather than `lon < 0`.
3. Use a different name for the variable `lon`; it refers to the difference of longitudes and the current name is somehow misleading.
4. Follow our coding rules and do not put spaces after `(` and before `)`.